jf_agent/main.py

```diff
@@ -233,15 +249,13 @@ def main():
                 )
 
             diagnostics.capture_outdir_size(config.outdir)
-            diagnostics.close_file()
-
-            # Kills the sys_diag_collector thread
-            sys_diag_done_event.set()
-            sys_diag_collector.join()
```

Comment:
we were doing this in multiple places that could/should just be handled by a `finally`